src/components/visualizations/chart/highcharts/heatmapConfiguration.ts

```diff
@@ -1,7 +1,57 @@
 // (C) 2007-2019 GoodData Corporation
 import cloneDeep = require("lodash/cloneDeep");
 import last = require("lodash/last");
+import { IHighchartsAxisExtend } from "../../../../interfaces/HighchartsExtend";
 
+interface IHeatmapSeries extends Highcharts.Series {
+    pointPlacementToXValue?(): number;
+}
+
+function between(x: number, a: number, b: number): number {
+    return Math.min(Math.max(a, x), b);
+}
+
+function resetPointPaddingForTooSmallHeatmapCells(series: IHeatmapSeries) {
+    const xAxis: IHighchartsAxisExtend = series.xAxis;
+    const yAxis: IHighchartsAxisExtend = series.yAxis;
+    const options: Highcharts.SeriesOptionsType = series.options;
+    const seriesPointPadding: number = options.pointPadding || 0;
+    const pointPlacement: number = series.pointPlacementToXValue();
+
+    series.points.forEach((point: Highcharts.Point) => {
+        // Recalculate (x1, x2, y1, y2) <=> (left, right, top, bottom) of heatmap cells in pixels
+        // Transform the calculation of x1, x2, y1, y2 from javascripts to typescripts
+        // From source: https://github.com/highcharts/highcharts/blob/v7.1.1/js/parts-map/HeatmapSeries.js#L222
+        const xPad: number = (options.colsize || 1) / 2;
+        const yPad: number = (options.rowsize || 1) / 2;
+        const x1 = between(
+            Math.round(xAxis.len - xAxis.translate(point.x - xPad, 0, 1, 0, 1, -pointPlacement)),
+            -xAxis.len,
+            2 * xAxis.len,
+        );
+        const x2 = between(
+            Math.round(xAxis.len - xAxis.translate(point.x + xPad, 0, 1, 0, 1, -pointPlacement)),
+            -xAxis.len,
+            2 * xAxis.len,
+        );
+        const y1 = between(
+            Math.round(yAxis.translate(point.y - yPad, 0, 1, 0, 1)),
+            -yAxis.len,
+            2 * yAxis.len,
+        );
+        const y2 = between(
+            Math.round(yAxis.translate(point.y + yPad, 0, 1, 0, 1)),
+            -yAxis.len,
+            2 * yAxis.len,
+        );
+        const pointPadding: number = point.pointPadding || seriesPointPadding;
+
+        if (Math.abs(x2 - x1) < pointPadding * 2 || Math.abs(y2 - y1) < pointPadding * 2) {
```

Comment:
If I am right, this is only part changed in comparison with original HCH calculation. Mark it by some more explicit comment

Reply:
comments is added.